src/Server/src/GraphQL/Client/TokenRequestDataResultFactory.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using IdOps.Abstractions;
+using IdOps.Model;
+using IdOps.Models;
+
+namespace IdOps.GraphQL;
+
+public class TokenRequestDataResultFactory : IResultFactory<TokenRequestData, TokenRequestInput>
+{
+    private readonly IEncryptionService _encryptionService;
+    private readonly IClientService _clientService;
+    private readonly IApiScopeService _scopeService;
+
+    public TokenRequestDataResultFactory(
+        IEncryptionService encryptionService,
+        IClientService clientService,
+        IApiScopeService apiScopeService)
+    {
+        _encryptionService = encryptionService;
+        _clientService = clientService;
+        _scopeService = apiScopeService;
+    }
+
+
+    public async Task<TokenRequestData> Create(TokenRequestInput input,
+        CancellationToken cancellationToken)
+    {
+        Client? client = await _clientService.GetByIdAsync(input.ClientId, cancellationToken);
+        if (client == null)
+        {
+            throw new ArgumentNullException($"Element with ID {input.ClientId} not found.");
+        }
+
+        var clientId = client.ClientId;
+
+        var secretEncrypted =
+            client.ClientSecrets.First(secret => secret.Id.Equals(input.SecretId)).EncryptedValue;
+        var secretDecrypted =
+            await _encryptionService.DecryptAsync(secretEncrypted, cancellationToken);
+
+        var grantTypes = client.AllowedGrantTypes.First();
```

Comment:
This looks wrong, what if there are many different GrantTypes on a client? Are we then just able to request a token for the first grant type in the list? Also what if the list is null or empty?